.github/workflows/go-tests.yml

```diff
@@ -49,9 +48,14 @@ jobs:
               echo "$files"
               exit 1
             fi
-      
+        
+      # Install gotestsum with go get for 1.15.3; otherwise default to go install 
       - name: Install gotestsum
-        run: go install gotest.tools/gotestsum@v1.8.2
+        run: |
+          if [ ${{ matrix.go-version }} == 1.15.3 ]; then
+              go get gotest.tools/gotestsum@v1.8.2; else
+              go install gotest.tools/gotestsum@v1.8.2
+          fi  
```

Comment:
This works, but is specific to go 1.15.3. I think I'd be tempted to use fallback to try to install this instead.. That means that we just try 'go install' first, and if that fails fall back to 'go test'. This should be resilient to additional versions being added to the matrix, so we'd have to update the code less often. There's also a check that whichever command succeeded we actually installed the binary into the path. This is because 'go get' might succeed in a later go version but wouldn't necessarily install the command to the system path.

Definitely feel free to reject this if you think it's overcomplicated btw! Your version works and is easier to read, I always tend to try & write code that won't need to change, but there are merits to both approaches.

```suggestion
        run: |
          GTS="gotest.tools/gotestsum@v1.8.2"
          # We use the same error message prefix in either failure case, so just define it once here.
          ERROR="Failed to install $GTS"
          # First try to 'go install', if that fails try 'go get'...
          go install "$GTS" || go get "$GTS" || { echo "$ERROR: both 'go install' and 'go get' failed"; exit 1; }
          # Check that the gotestsum command was actually installed in the path...
          command -v gotestsum > /dev/null 2>&1 || { echo "$ERROR: gotestsum command not installed"; exit 1; }
          echo "OK: Command 'gotestsum' installed ($GTS)"
```

Reply:
I really like this suggestion, thank you! I'm all for more robustability-- I learned something new!